zeroex/orderwatch/order_watcher_utils.go

```diff
@@ -0,0 +1,41 @@
+package orderwatch
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type contractAddressesSeenCounter struct {
+	contractAddressToSeenCount map[common.Address]uint
+
+	mu sync.RWMutex
```

Comment:
This is a good place for atomic counters [`sync/atomic`](https://golang.org/pkg/sync/atomic/) instead of implementing our own using locks. They have some performance advantages under the hood, but I don't think that matters here.

Reply:
Then again, we still need a lock for map itself if entries get added, physically moved or removed.